pkg/conversion/capi2mapi/powervs.go

```diff
@@ -0,0 +1,305 @@
+		template:       mts,
+		powerVSCluster: pc,
+		machineAndPowerVSMachineAndPowerVSCluster: &machineAndPowerVSMachineAndPowerVSCluster{
+			machine: &capiv1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      ms.Spec.Template.ObjectMeta.Labels,
+					Annotations: ms.Spec.Template.ObjectMeta.Annotations,
+				},
+				Spec: ms.Spec.Template.Spec,
+			},
+			powerVSMachine: &capibmv1.IBMPowerVSMachine{
+				Spec: mts.Spec.Template.Spec,
+			},
+			powerVSCluster: pc,
+		},
+	}
+}
+
+// ToMachine converts a capi2mapi MachineAndPowerVSMachineTemplate into a MAPI Machine.
+func (m machineAndPowerVSMachineAndPowerVSCluster) ToMachine() (*mapiv1beta1.Machine, []string, error) {
+	if m.machine == nil || m.powerVSMachine == nil || m.powerVSCluster == nil {
+		return nil, nil, errCAPIMachinePowerVSMachinePowerVSClusterCannotBeNil
+	}
+
+	var (
+		errors   []error
+		warnings []string
+	)
+
+	mapiPowerVSSpec, warn, err := m.toProviderSpec()
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	warnings = append(warnings, warn...)
+
+	mapiMachine, err := fromCAPIMachineToMAPIMachine(m.machine)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	powerVSRawExt, err := rawExtensionFromPowerVSProviderSpec(mapiPowerVSSpec)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	mapiMachine.Spec.ProviderSpec.Value = powerVSRawExt
+
+	if len(errors) > 0 {
+		return nil, warnings, utilerrors.NewAggregate(errors)
+	}
+
+	return mapiMachine, warnings, nil
+}
+
+// ToMachineSet converts a capi2mapi MachineAndPowerVSMachineTemplate into a MAPI MachineSet.
+func (m machineSetAndPowerVSMachineTemplateAndPowerVSCluster) ToMachineSet() (*mapiv1beta1.MachineSet, []string, error) {
+	if m.machineSet == nil || m.template == nil || m.powerVSCluster == nil || m.machineAndPowerVSMachineAndPowerVSCluster == nil {
+		return nil, nil, errCAPIMachineSetPowerVSMachineTemplatePowerVSClusterCannotBeNil
+	}
+
+	var (
+		errors   []error
+		warnings []string
+	)
+
+	// Run the full ToMachine conversion so that we can check for
+	// any Machine level conversion errors in the spec translation.
+	mapiPowerVSMachine, warn, err := m.ToMachine()
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	warnings = append(warnings, warn...)
+
+	mapiMachineSet, err := fromCAPIMachineSetToMAPIMachineSet(m.machineSet)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	mapiMachineSet.Spec.Template.Spec = mapiPowerVSMachine.Spec
+
+	// Copy the labels and annotations from the Machine to the template.
+	mapiMachineSet.Spec.Template.ObjectMeta.Annotations = mapiPowerVSMachine.ObjectMeta.Annotations
+	mapiMachineSet.Spec.Template.ObjectMeta.Labels = mapiPowerVSMachine.ObjectMeta.Labels
+
+	if len(errors) > 0 {
+		return nil, warnings, utilerrors.NewAggregate(errors)
+	}
+
+	return mapiMachineSet, warnings, nil
+}
+
+// toProviderSpec converts a capi2mapi machineAndPowerVSMachineAndPowerVSCluster into a MAPI PowerVSMachineProviderConfig.
+//
+//nolint:funlen
```

Comment:
Is this needed? The function doesn't look super long?

Reply:
Its not needed.